mesonbuild/interpreter.py

```diff
@@ -1932,17 +1933,19 @@ def override_dependency_method(self, args, kwargs):
         name = args[0]
         dep = args[1]
         if not isinstance(name, str) or not name:
-            raise InterpreterException('First argument must be not empty string')
+            raise InterpreterException('First argument must be a string and cannot be empty')
         if hasattr(dep, 'held_object'):
             dep = dep.held_object
         if not isinstance(dep, dependencies.Dependency):
             raise InterpreterException('Second argument must be a dependency object')
         identifier = dependencies.get_dep_identifier(name, kwargs)
         for_machine = self.interpreter.machine_from_native_kwarg(kwargs)
-        if identifier in self.build.dependency_overrides[for_machine]:
-            raise InterpreterException('Tried to override dependency "%s" which has already been overridden.'
-                                       % name)
-        self.build.dependency_overrides[for_machine][identifier] = dep
+        override = self.build.dependency_overrides[for_machine].get(identifier)
+        if override:
+            m = 'Tried to override dependency {!r} which has already been looked up or overridden at {}'
```

Comment:
s/looked up/resolved/?

i.e. finding / searching / lookup is something that can yield a non fatal "not found" but resolve is the process when meson creates an immutable binding from name to the actual dependency object.

Reply:
Doesn't change much, but why not. Done.